src/fsharp/FSharp.Core/array.fs

```diff
@@ -74,8 +74,12 @@ namespace Microsoft.FSharp.Collections
             if array.Length = 0 then invalidArg "array" (SR.GetString(SR.notEnoughElements))            
             Microsoft.FSharp.Primitives.Basics.Array.subUnchecked 1 (array.Length - 1) array
 
+        type EmptyArray<'T>() = 
```

Comment:
Could you do this as a separate PR please?  It could have wide reaching impact and we might not take it in F# 4.5. Thanks

Reply:
In particular I don't really want a situation where people will use `Array.empty` instead of `[| |]` just to get the singleton.  So doing the work in F# compiler to make `[| |]` call `Array.empty` should be part of adding this.  And we should assess the overall perf impact too, and also make sure the functions like `Array.map` return the singleton if the input was empty,

Reply:
Doing it in a separate PR will allow us to address all these things at once (though in F# 5.0, not 4.5).  Or at least allows us to discuss that